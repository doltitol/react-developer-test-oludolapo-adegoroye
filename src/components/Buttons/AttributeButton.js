import React, { PureComponent } from 'react';

export class AttributeButton extends PureComponent {
  render() {
    return (
      <button
        className='attribute-button'
        style={{
          background: this.props.active ? '#1D1F22' : 'transparent',
          width: this.props.small ? '32px' : '63px',
          height: this.props.small ? '28px' : '45px',
          color: this.props.active ? '#FFFFFF' : '#1D1F22',
          fontSize: this.props.small ? '0.6em' : '1.1em',
        }}
        onClick={this.props.onClick}
      >
        {this.props.text}
      </button>
    );
  }
}

export default AttributeButton;
