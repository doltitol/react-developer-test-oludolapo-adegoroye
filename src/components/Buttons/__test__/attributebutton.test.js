import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AttributeButton from '../AttributeButton';

describe('<AttributeButton /> spec', () => {
  it('renders the component', () => {
    const container = render(<AttributeButton text='s' active={true} />);
    const getByTestId = container.getByTestId;
    const attributeButton = getByTestId('attribute-button');
    expect(attributeButton.textContent).toEqual('s');
    expect(attributeButton).toHaveStyle('background: #1D1F22');
    expect(attributeButton).toHaveStyle('color: #FFFFFF');
    expect(container.children).toMatchSnapshot();
  });
});
