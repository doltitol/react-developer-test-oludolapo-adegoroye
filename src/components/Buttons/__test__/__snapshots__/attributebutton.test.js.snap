// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttributeButton /> spec renders the component 1`] = `undefined`;

exports[`<MainButton /> spec renders the component 1`] = `undefined`;
